neonvm/runner/main.go

```diff
@@ -925,7 +934,9 @@ func forwardLogs(ctx context.Context, logger *zap.Logger, wg *sync.WaitGroup) {
 			if conn != nil {
 				conn.Close()
 			}
-			_ = drainLogsReader(reader, logger)
+			if reader != nil {
+				_ = drainLogsReader(reader, logger)
+			}
```

Comment:
This could cause us to drop logs if the VM shuts down very quickly after starting up, right?

Can we try dialing the socket if `reader == nil` to possibly recover?

Reply:
oh, QEMU removes the socket when it's done? or is it that you only can receive from the socket when there's someone else writing

Reply:
> QEMU removes the socket when it's done?

Yes. Socket is a "part" of virtio-serial device, so it's lifecycle is tied to it.